### PR TITLE
Push public key to project if not already present

### PR DESCRIPTION
**Pull request by fejta**
Fixes https://github.com/kubernetes/kubernetes/issues/26128
Fixes https://github.com/kubernetes/kubernetes/issues/26129

Whenever the ssh key changes (which happens right now whenever we add/change a new jenkins agent) this key will not get pushed to the project until either `gcloud compute ssh` or `gcloud compute config-ssh` runs. As a result instances on this project will reject ssh attempts with this key.

In many cases this will not happen until after a test on a project fails and we attempt to ssh to the nodes to grab logs.

This verifies the presence of the ssh key before starting tests, and attempts to add it if it is missing.

### CODE REVIEW

File: hack/jenkins/e2e-runner.sh [hunk 1]
```diff
@@ -257,12 +257,19 @@ case "${KUBERNETES_PROVIDER}" in
...
-        if [[ ! -f ${WORKSPACE}/.ssh/google_compute_engine ]]; then
-            echo "google_compute_engine ssh key missing!"
+        echo 'Checking existence of private ssh key'
+        local _key="${WORKSPACE}/.ssh/google_compute_engine"
+        if [[ ! -f "${_key}" || ! -f "${_key}.pub" ]]; then
+            echo 'google_compute_engine ssh key missing!'
             exit 1
         fi
+        echo "Checking presence of public key in ${PROJECT}"
+        if ! gcloud compute "--project=${PROJECT}" project-info describe |
```
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: I'm wary of `gcloud compute` piped into `grep`. Is it possible to use a format flag to ensure that we get exactly what we want, even if the default format changes?

Comment (fejta) [inline hack/jenkins/e2e-runner.sh]: I would agree were I trying to parse the output, but I'm just trying to ensure the public key exists.

I would rather avoid trying to process the json output and validate that
```
output = json.loads(gcloud_output)
for key in output['commonInstanceMetadata']['items']['sshKeys'].split('\n'):
  if public_key in key: return True
else:
  return False
```

Seems overkill here (and annoying to do in bash)

File: hack/jenkins/e2e-runner.sh [hunk 2]
```diff
@@ -257,12 +257,19 @@ case "${KUBERNETES_PROVIDER}" in
             cp /var/lib/jenkins/gce_keys/google_compute_engine ${WORKSPACE}/.ssh/
             cp /var/lib/jenkins/gce_keys/google_compute_engine.pub ${WORKSPACE}/.ssh/
         fi
-        if [[ ! -f ${WORKSPACE}/.ssh/google_compute_engine ]]; then
-            echo "google_compute_engine ssh key missing!"
+        echo 'Checking existence of private ssh key'
+        local _key="${WORKSPACE}/.ssh/google_compute_engine"
```
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: nit: I don't think we use underscores to prefix variables anywhere else, so I'd drop it for consistency.

File: hack/jenkins/e2e-runner.sh (hunk 1 above)
Comment (spxtr) [inline hack/jenkins/e2e-runner.sh]: Seems reasonable to me.

Nit: I think it makes more sense to say `--project="${PROJECT}"`.

**Comment by spxtr** [comment]
LGTM other than nits.

**Comment by quinton-hoole** [comment]
Added P0 and v1.3 to get this merged ASAP, as this is causing multiple e2e tests to fail, and contributing to merge queue stalls.

**Pull request merged**